plugins/inputs/opcua_listener/integration_test.go

```diff
@@ -0,0 +1,151 @@
+package opcua_listener
+
+import (
+	"context"
+	"fmt"
+	"github.com/docker/go-connections/nat"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/opcua"
+	"github.com/influxdata/telegraf/plugins/common/opcua/input"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"reflect"
+	"testing"
+	"time"
+)
+
+const servicePort = "4840"
+
+type OPCTags struct {
+	Name           string
+	Namespace      string
+	IdentifierType string
+	Identifier     string
+	Want           interface{}
+}
+
+func MapOPCTag(tags OPCTags) (out input.NodeSettings) {
+	out.FieldName = tags.Name
+	out.Namespace = tags.Namespace
+	out.IdentifierType = tags.IdentifierType
+	out.Identifier = tags.Identifier
+	return out
+}
+
+func TestSubscribeClientIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+
+	container := testutil.Container{
+		Image:        "open62541/open62541",
+		ExposedPorts: []string{servicePort},
+		WaitingFor:   wait.ForListeningPort(nat.Port(servicePort)),
+	}
+	err := container.Start()
+	require.NoError(t, err, "failed to start container")
+	defer func() {
+		require.NoError(t, container.Terminate(), "terminating container failed")
+	}()
+
+	var testopctags = []OPCTags{
+		{"ProductName", "0", "i", "2261", "open62541 OPC UA Server"},
+		{"ProductUri", "0", "i", "2262", "http://open62541.org"},
+		{"ManufacturerName", "0", "i", "2263", "open62541"},
+		{"badnode", "1", "i", "1337", nil},
+		{"goodnode", "1", "s", "the.answer", "42"},
+	}
+	var tagsRemaining = make([]string, 0, len(testopctags))
+	for i, tag := range testopctags {
+		if tag.Want != nil {
+			tagsRemaining = append(tagsRemaining, testopctags[i].Name)
+		}
+	}
+
+	subscribeConfig := SubscribeClientConfig{
+		InputClientConfig: input.InputClientConfig{
+			OpcUAClientConfig: opcua.OpcUAClientConfig{
+				Endpoint:       fmt.Sprintf("opc.tcp://%s:%s", container.Address, container.Ports[servicePort]),
+				SecurityPolicy: "None",
+				SecurityMode:   "None",
+				AuthMethod:     "Anonymous",
+				ConnectTimeout: config.Duration(10 * time.Second),
+				RequestTimeout: config.Duration(1 * time.Second),
+				Workarounds:    opcua.OpcUAWorkarounds{},
+			},
+			MetricName: "testing",
+			RootNodes:  make([]input.NodeSettings, 0),
+			Groups:     make([]input.NodeGroupSettings, 0),
+		},
+		SubscriptionInterval: 0,
+	}
+	for _, tags := range testopctags {
+		subscribeConfig.RootNodes = append(subscribeConfig.RootNodes, MapOPCTag(tags))
+	}
+	o, err := subscribeConfig.CreateSubscribeClient(testutil.Logger{})
```

Comment:
The linter complains here because there is no error handling done. Either implement error handling or do like this to ignore any errors:
```suggestion
	o, _ := subscribeConfig.CreateSubscribeClient(testutil.Logger{})
```

Reply:
Thank you! Interesting, GoLand didn't mark it. It's usually pretty good with these kinds of things.

Reply:
Where did you find the line number with the error? In the logs I didn't see it.